src/main/java/htsjdk/tribble/gff/Gff3Writer.java

```diff
@@ -0,0 +1,103 @@
+package htsjdk.tribble.gff;
+
+import htsjdk.samtools.util.FileExtensions;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.tribble.TribbleException;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+
+public class Gff3Writer implements Closeable {
+
+    private final PrintStream out;
+    private final static String version = "3.1.25";
+
+    public Gff3Writer(final Path path) throws IOException {
+        if (!FileExtensions.GFF3.stream().anyMatch(e -> path.toString().endsWith(e))) {
+            throw new TribbleException("File " + path + " does not have extension consistent with gff3");
+        }
+
+        final OutputStream outputStream = IOUtil.hasGzipFileExtension(path)? new GZIPOutputStream(Files.newOutputStream(path)) : Files.newOutputStream(path);
```

Comment:
Avoid GZIPOutputStream.  GZIP streams in java are moderately cursed.  If you really mean GZIP and not BGZIP you can get a stream with IOUtil.openGzipFileForWriting.  

Are these files not indexable in any way?  I can't remember... Are they not sorted?  If they are meant to be randomly accessible you should really be using a bgzip output.

Reply:
They can be sorted, in which case they can be indexed.  Changed to bgzip